docs/tutorials/03_minimum_eigen_optimizer.ipynb

```diff
@@ -293,6 +297,270 @@
     "print(qaoa_result)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Analysis of Samples\n",
+    "Objects of type `SolutionSample` (here denoted as *samples*) are a useful information source provided\n",
+    "by `OptimizationResult` objects. They contain information about input values `x`, corresponding (optimized) function\n",
+    "values `fval`, probability of obtaining that result `probability` and the solution status `status` (`SUCCESS`,\n",
+    "`FAILURE`, `INFEASIBLE`).\n",
+    "\n",
+    "They can be easily printed out in a better-readable format."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 48,
+   "outputs": [],
+   "source": [
+    "def print_samples(samples: List[SolutionSample],\n",
+    "                  max_len: int = None):\n",
+    "    status_str = {OptimizationResultStatus.SUCCESS: 'SUCCESS',\n",
+    "                  OptimizationResultStatus.FAILURE: 'FAILURE',\n",
+    "                  OptimizationResultStatus.INFEASIBLE: 'INFEASIBLE'}\n",
+    "    print('\\n'.join(['[{}]: fval: {:.4f}, probability: {:.2f} [{}]'.format(''.join(str(int(_)) for _ in e.x),\n",
+    "                                                                           e.fval,\n",
+    "                                                                           e.probability,\n",
+    "                                                                           status_str[e.status])\n",
+    "            for e in sorted(samples, key=lambda x: x.probability, reverse=True)][:max_len]))"
```

Comment:
Please do not sort the samples by yourself here. qiskit-opt sorts the samples by the order  of the objective function value.

Reply:
I don't think you need to define a function. The following snippet is enough.
```
for sample in samples:
    print(sample)
```